File: Python_Codes/12. Enums & Annotations/EnumSwitchCase.py
from enum import Enum

# Enum class for seasons
class Season(Enum):
    SPRING = 1
    SUMMER = 2
    FALL = 3
    WINTER = 4

# Main logic
def main():
    current_season = Season.WINTER

    # Using if-elif-else instead of match-case
    if current_season == Season.SPRING:
        print("Spring is blooming with flowers!")
    elif current_season == Season.SUMMER:
        print("Summer is hot and sunny!")
    elif current_season == Season.FALL:
        print("Fall brings beautiful autumn leaves!")
    elif current_season == Season.WINTER:
        print("Winter calls for warm clothes!")
    else:
        print("Unknown season.")

if __name__ == "__main__":
    main()

File: Projects/JavaProject/src/com/Enums_And_Annotations/EnumSwitchCase.java
package com.Enums_And_Annotations;

// Enum class for seasons
enum Season {
    SPRING, SUMMER, FALL, WINTER;
}

// Main class to demonstrate Switch Case
public class EnumSwitchCase {
    public static void main(String[] args) {
        // Example usage of Enum with switch case
        Season currentSeason = Season.WINTER;

        // Print messages based on the season
        switch (currentSeason) {
            case SPRING:
                System.out.println("Spring is blooming with flowers!");
                break;
            case SUMMER:
                System.out.println("Summer is hot and sunny!");
                break;
            case FALL:
                System.out.println("Fall brings beautiful autumn leaves!");
                break;
            case WINTER:
                System.out.println("Winter calls for warm clothes!");
                break;
            default:
                System.out.println("Unknown season.");
        }
    }
}
